python/cudf/cudf/core/df_protocol.py

```diff
@@ -89,7 +89,7 @@ def ptr(self) -> int:
         """
         Pointer to start of the buffer as an integer.
         """
-        return self._buf.ptr
+        return self._buf.get_ptr(mode="write")
```

Comment:
This is because the dataframe interchange protocol is a zero-copy one, so an external consumer is getting a view?

Reply:
I'm not quite familiar with df interchange protocol if it has to be a zero-copy one. I just made an equivalent switch of `.ptr` to `mode='write` assuming the buffer should be marked as exposed at this point since we are handing off to another library, my understanding here can be wrong too. @shwina to confirm.

Reply:
There's no promise that the interchange protocol is zero-copy. In-fact, in the case of cross-device (CPU<->GPU) interchange, it cannot be.

That being said, I think any public API via which we expose the raw pointer to a buffer should mark the buffer exposed.

Reply:
Thanks for clarifying @shwina! This will mark the buffer as exposed.